tests/agent_server/deploy/e2e/test_autostarted.py

```diff
@@ -244,3 +253,750 @@ def repaired() -> bool:
+    assert (
+        len(current_children_after_deployment) == expected_additional_children_after_deployment
+    ), "These processes should be present: The Scheduler, the fork server and the actual agent!"
+    for children in current_children_after_deployment.values():
+        assert children.is_running()
+
+    # The number of resources for this version should match
+    result = await client.list_versions(tid=environment)
+    assert result.code == 200
+    assert len(result.result["versions"]) == 1
+    assert result.result["versions"][0]["total"] == 3
+
+    # Let's check the agent table and check that agent1 is present and not paused
+    result = await client.list_agents(tid=environment)
+    assert result.code == 200
+    assert len(result.result["agents"]) == 2
+    expected_agents_status = {e["name"]: e["paused"] for e in result.result["agents"]}
+    assert set(expected_agents_status.keys()) == {const.AGENT_SCHEDULER_ID, "agent1"}
+    assert not expected_agents_status[const.AGENT_SCHEDULER_ID]
+    assert not expected_agents_status["agent1"]
+
+    # Now let's pause agent1
+    result = await client.agent_action(tid=environment, name="agent1", action=AgentAction.pause.value)
+    assert result.code == 200
+
+    # Let's recheck the agent table and check that agent1 is present and paused
+    result = await client.list_agents(tid=environment)
+    assert result.code == 200
+    assert len(result.result["agents"]) == 2
+    expected_agents_status = {e["name"]: e["paused"] for e in result.result["agents"]}
+    assert set(expected_agents_status.keys()) == {const.AGENT_SCHEDULER_ID, "agent1"}
+    assert not expected_agents_status[const.AGENT_SCHEDULER_ID]
+    assert expected_agents_status["agent1"]
+
+    # Let's also check if the state of resources are consistent with what we expect: one resource still needs to be
+    # deployed
+    await retry_limited(are_resources_deployed, timeout=6, deployed_resources=2)
+    result = await client.resource_list(environment, deploy_summary=True)
+    assert result.code == 200
+    summary = result.result["metadata"]["deploy_summary"]
+    assert summary["total"] == 3, f"Unexpected summary: {summary}"
+    assert summary["by_state"]["available"] == 1, f"Unexpected summary: {summary}"
+    assert summary["by_state"]["deployed"] == 2, f"Unexpected summary: {summary}"
+
+    # Let's wait for the executor to die
+    await retry_limited(
+        wait_for_terminated_status, timeout=6, interval=1, current_children=current_children_after_deployment.values()
+    )
+
+    # Let's recheck the number of processes after pausing the agent
+    halted_children = get_process_state(current_pid)
+    assert len(halted_children) == 1
+    assert len(halted_children.values()) == 1
+    current_halted_children = filter_relevant_processes(halted_children)
+
+    # We should have lost one process in the "process"
+    assert len(current_halted_children) == len(current_children_after_deployment) - 1
```

Comment:
Why? All these agent share an executor process  and filter_relevant_processes collapse all executors into one entry anyways. So you just wait for the executor cleanup, which has no relation to halting.